Issues discovered by mypy

$ `mypy --ignore-missing-imports .`
```
first_lesson/input.py:39: error: Incompatible types in assignment (expression has type "int", variable has type "str")
first_lesson/SortArray.py:24: error: Missing return statement
```

--- a/.github/workflows/lint_python.yml
+++ b/.github/workflows/lint_python.yml
@@ -7,13 +7,13 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
       - run: pip install bandit black codespell flake8 isort mypy pytest pyupgrade safety
-      - run: bandit --recursive --skip B101 . || true  # B101 is assert statements
+      - run: bandit --recursive --skip B101 .  # B101 is assert statements
       - run: black --check . || true
-      - run: codespell --quiet-level=2 || true  # --ignore-words-list="" --skip=""
+      - run: codespell --quiet-level=2  # --ignore-words-list="" --skip=""
       - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
       - run: isort --check-only --profile black . || true
       - run: pip install -r requirements.txt || true
-      - run: mypy --ignore-missing-imports . || true
+      - run: mypy --ignore-missing-imports .
       - run: pytest . || true
       - run: pytest --doctest-modules . || true
       - run: shopt -s globstar && pyupgrade --py36-plus **/*.py || true

--- a/.github/workflows/lint_python.yml
+++ b/.github/workflows/lint_python.yml
@@ -1,0 +1,20 @@
+name: lint_python
+on: [pull_request, push]
+jobs:
+  lint_python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: pip install bandit black codespell flake8 isort mypy pytest pyupgrade safety
+      - run: bandit --recursive --skip B101 . || true  # B101 is assert statements
+      - run: black --check . || true
+      - run: codespell --quiet-level=2 || true  # --ignore-words-list="" --skip=""
+      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: isort --check-only --profile black . || true
+      - run: pip install -r requirements.txt || true
+      - run: mypy --ignore-missing-imports . || true
+      - run: pytest . || true
+      - run: pytest --doctest-modules . || true
+      - run: shopt -s globstar && pyupgrade --py36-plus **/*.py || true
+      - run: safety check

--- a/first_lesson/SortArray.py
+++ b/first_lesson/SortArray.py
@@ -1,4 +1,4 @@
-#!/bin/env/python3
+#!/usr/bin/env python3
 
 '''
 Problem:

--- a/first_lesson/SortArray.py
+++ b/first_lesson/SortArray.py
@@ -21,32 +21,32 @@ psudocode:
 
 #code
 
-def sortDatainAccending(array: list)->list:
+def sortDatainAccending(array: list) -> list:
     try:
-    
         print()
         print(f"Data after sorting{array}")
         '''
-        This funcation takes an array as parameter
+        This function takes an array as parameter
         Example :
         sortDatainAccending([99,20,34,65,78])
         --------------------------------------
-        (Funcation name ( [parameter] :TYPE list  )   )
+        (Function name ( [parameter] :TYPE list  )   )
         '''
         length = len(array) # This will find the length of an array.
         sortArray = [] # This empty array stores the sorted result.
         for i in range(length):
-            
             minData = min(array) # Find the min value from array.
             sortArray.append(minData) # Appending Result in new sorted array.
-            array.remove(minData) # Removing minData from array that alrady is beging checked or sorted form previous list.
+            array.remove(minData) # Removing minData from array that already is beging checked or sorted form previous list.
         print(f"The Sorted data is : {sortArray}")
         print()
-    except:
-        print("Sorting opretion cannot be done in <int> and <string> data")
+        return sortArray
+    except TypeError:
+        print("Sorting operation cannot be done in <int> and <string> data")
+        return []
         
-sortDatainAccending([20,34,5000,23,1,0,0.1,-8,-1,23])
-sortDatainAccending(['a','z','c','abc'])
-sortDatainAccending(['abc',12,'abc'])
+print(sortDatainAccending([20,34,5000,23,1,0,0.1,-8,-1,23]))
+print(sortDatainAccending(['a','z','c','abc']))
+print(sortDatainAccending(['abc',12,'abc']))
 
 # code :- github.com/pandatd/

--- a/first_lesson/helloworld.py
+++ b/first_lesson/helloworld.py
@@ -1,21 +1,14 @@
-#!/bin/python3.9 
+#!/usr/bin/env python3.9 
 
 '''
 Note:-
-    There are major two versions of python 
-    that is python 2.x  and  python 3.x 
-    Now in this all following tutorial we will be using 
-    python 3.x 
+    The following tutorial we will be using Python 3.9
     
     How to use ?
     
-    If you want print the string or somethig 
-    you must have to use print funcation which will bultine 
-    in python 3.x version 
+    If you want print the string or something  you have to
+    use the print() function which will builtin in Python 3.x.
     
     '''
-# Syntax in python 3.x 
+# Syntax in Python 3.x 
 print(" Hello world ") 
-
-# Syntax in python 2.x
-# print "Hello World " # Note we are not using this in for following all tutorials 

--- a/first_lesson/helloworld.py
+++ b/first_lesson/helloworld.py
@@ -6,7 +6,7 @@ Note:-
     
     How to use ?
     
-    If you want print the string or something  you have to
+    If you want print a string or something else you have to
     use the print() function which will builtin in Python 3.x.
     
     '''

--- a/first_lesson/input.py
+++ b/first_lesson/input.py
@@ -5,7 +5,7 @@ syntax :-
         object = function - message
         the taken input is being stored in object variable
         don't be upset we will dissuss more about variavles in
-        upcomming leactures
+        upcoming leactures
 example :-
         my_name = input("Enter Your Name : )
         print(my_name)
@@ -18,7 +18,7 @@ output  :-
         A) Today , we will learn about how to take input in Python
         1. The input function will take  string parameter by default
         2. Yes, we can take integer input also , but we have to pass in
-           int() funcation. '''
+           int() function. '''
 
 input("Let's start lesson... Press any key to start")
 my_name = input("Enter Your Name : ")

--- a/first_lesson/input.py
+++ b/first_lesson/input.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 note = '''
 input()
 syntax :-

--- a/first_lesson/input.py
+++ b/first_lesson/input.py
@@ -2,7 +2,7 @@ note = '''
 input()
 syntax :-
         object = input("<Message to display on screen>")
-        object  = funcation - message
+        object = function - message
         the taken input is being stored in object variable
         don't be upset we will dissuss more about variavles in
         upcomming leactures
@@ -13,14 +13,14 @@ output  :-
         Enter Your Name : Tejas Dixit
         Tejas Dixit
 
-        please don't include the <> symbols ,
-        in python you can use double or single quotes to print.
-        A) Today , we will learn about how to take input in python
-        1. The input funcation will take  string parameter by deafult
+        Please don't include the <> symbols ,
+        in Python you can use double or single quotes to print.
+        A) Today , we will learn about how to take input in Python
+        1. The input function will take  string parameter by default
         2. Yes, we can take integer input also , but we have to pass in
            int() funcation. '''
 
-input("Lets start lesson Press any key to start")
+input("Let's start lesson... Press any key to start")
 my_name = input("Enter Your Name : ")
 print(my_name)
 type_of_data = type(my_name)
@@ -36,12 +36,12 @@ code = '''my_age = int(input("Enter Your Age: "))
           print(my_age)
           print("myage is ",type(my_age),"type of data")'''
 
-my_age = int(input("Enter Your Age: "))
+my_age = int(input("Enter Your Age: "))  # type: ignore
 print(my_age)
 print("myage is ",type(my_age),"type of data")
 print("Now it shows int confused how is it happen.")
 input("For Explenetion press any key")
 print("code of following example")
 print(code)
-print("Explenetion: ")
+print("Explanation: ")
 print(note)

